src/app/components/kudos-transactions-stream/kudos-transactions-stream.directive.js

```diff
@@ -0,0 +1,125 @@
+/**
+ * Created by vytautassugintas on 25/04/16.
+ */
+angular.module('myApp.components', [])
+    .directive('kudosTransactionsStream', function () {
+        return {
+            controller: 'KudosTransactionController',
+            restrict: 'E',
+            scope: false,
+            templateUrl: 'app/components/kudos-transactions-stream/kudos-transactions-stream.html'
+        }
+    })
+    .controller('KudosTransactionController', function ($scope, $timeout, $httpParamSerializer, ProfileService, KudosTransactionService) {
+
+        var page = 0;
+        var pageSize = 10;
+
+        var requestData = $httpParamSerializer({
+            page: page,
+            pageSize: pageSize
+        });
+
+        $scope.transactionCollection = [];
+
+        $scope.showMoreTransactions = showMoreTransactions;
+        $scope.showLessTransactions = showLessTransactions;
+        $scope.acornPlural = acornPlural;
+        $scope.getKudosTransactionsFeed = getKudosTransactionsFeed;
+        $scope.changeTransactionsList = changeTransactionsList;
+
+        getKudosTransactionsFeed();
```

Comment:
Use activate() f-tion as in John Papas styleguide.

Or if you on 1.5.x angular write components instead of controllers and then use $onInit